praw/models/reddit/collections.py

```diff
@@ -354,12 +355,12 @@ def __init__(
         reddit: praw.Reddit,
         subreddit: praw.models.Subreddit,
         _data: dict[str, Any] | None = None,
-    ):
+    ) -> None:
         """Initialize a :class:`.SubredditCollections` instance."""
         super().__init__(reddit, _data)
         self.subreddit = subreddit
 
-    def __iter__(self):
+    def __iter__(self) -> Generator[Collection, None, None]:
```

Comment:
Would this be an `Iterator`?

Reply:
Yes. I'll look to simplify these Generator type definitions to Iterator.